### PR TITLE
docs(app-of-apps): argoconfigs template

### PR DESCRIPTION
**Pull request by hairmare**
# Description

Add a section on the `argoconfig` lilbrary chart.

# Issues

- https://github.com/adfinis-sygroup/helm-charts/issues/84

# Checklist

* [x] ~~I updated the version in Chart.yaml~~
* [x] ~~I updated applicable README.md files using  `pre-commit run`~~
* [x] I documented any high-level concepts I'm introducing in `docs/`
* [x] ~~If I updated a dependency tool, or app, this PR contains a short summary of the changes I'm pulling~~
* [x] CI is currently green and this is ready for review
* [x] ~~I am ready to test changes after they are applied and released~~

### CODE REVIEW

**Review by tongpu — COMMENTED** [review]
I'll have to read the second part again tomorrow, but here are some minor recommendations from my side regarding how to lead over to code snippets. I would use `:`, but I'm actually not sure if the English language might be different from German in this regard.

File: docs/argocd-app-of-apps.md [hunk 1]
```diff
@@ -26,6 +26,45 @@ we consider valuable on most clusters. As with all the charts, `infra-apps` can
...
+# ... common.metadata
+spec:
+  project: "default"
+  source: {}
+  destination:
+    server: "https://kubernetes.default.svc"
+  syncPolicy: {}
+```
+
+Our app-of-apps charts invoke the function in their individual `Application` templates:
```
Comment (tongpu) [inline docs/argocd-app-of-apps.md:48]: After a nights sleep I'm now able to phrase what was bothering me yesterday. What I'm missing here is the information how to configure an application contained in an app-of-apps chart, e.g. how does the structure look like in `values.yaml` and what needs to be configured where.

This could possibly be a topic on top of the `argoconfig` section, because it would allow the reader to read on from a usage documentation to the exact implementation.

Do you understand what I'm aiming for?

Comment (hairmare) [reply docs/argocd-app-of-apps.md:48]: see b89b794df89b51c4c3ac0203ae45aee005aeb64c

File: docs/argocd-app-of-apps.md [hunk 2]
```diff
@@ -26,6 +26,57 @@ we consider valuable on most clusters. As with all the charts, `infra-apps` can
...
+  {{- end }}
+  # ... more overriding, check any app-of-apps chart for an example
+{{- end -}}
+```
+
+The first block in this example renders the `Application`, the second block injects a template used for rendering. The first thing that
+is done in the second block is to assign everything specific to an app to `$app` so it can be referenced in the template without needing
+to write out the full `.Values.example` part over and over.
+
+In the above example the values section would contain a minimal example section as default:
```
Comment (tongpu) [inline docs/argocd-app-of-apps.md:70]:
```suggestion
In the above example the `values.yaml` for an app-of-apps chart would contain a minimal section for an `example` application by default:
```

File: docs/argocd-app-of-apps.md [hunk 3]
```diff
@@ -26,6 +26,57 @@ we consider valuable on most clusters. As with all the charts, `infra-apps` can
...
+
+```yaml
+example:
+  enabled: false
+  name: example
+  # ... more values, check any app-of-apps chart for an example
+  value: {}
```
Comment (tongpu) [inline docs/argocd-app-of-apps.md:77]:
```suggestion
  values: {}
```

**Review by tongpu — COMMENTED** [review]
I think we're still not at the point where we're able to really explain the concept in layman's terms, but I'm not sure if this is even possible.

**Comment by hairmare** [comment]
> I think we're still not at the point where we're able to really explain the concept in layman's terms, but I'm not sure if this is even possible.

This really only focuses on the library chart bits, #84 has some more tasks that should get us closer to explaning the big picture concept.

**Review by tongpu — APPROVED** [review]
